### PR TITLE
Review Security Checks Docs

### PR DESCRIPTION
**Pull request by josemigallas**
JIRA: https://issues.jboss.org/browse/AEROGEAR-3331

TODO:
- [x] Index page
- [x] Coding
- [x] Device Trust Checks
    - [x] Developer Mode Detection for iOS (is it not available? ping @StephenCoady )
    - [x] Developer Mode Detection for Cordova (is it not available? pinging @wtrocki )
    - [x] Reporting check results to Metrics Service for Xamarin (pinging @ziccardi )
- [x] Certificate Pinning

### CODE REVIEW

File: modules/ROOT/pages/security/coding.adoc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+[role="primary"]
+.Android
+
+****
+
+. Add `jcenter` to your project's *build.grade*:
++
+[source, groovy]
+----
+allprojects {
```
Comment (pb82) [inline modules/ROOT/pages/security/coding.adoc]: that part is already explained in `build/site/aerogear/latest/configuring-dev-env.html`. Maybe just add that read as a prereq?

File: modules/ROOT/pages/security/coding.adoc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+[role="secondary"]
+.Xamarin
+
+****
+
+. Install the link:https://www.nuget.org/packages/AeroGear.Mobile.Security[AeroGear Security] package from link:https://docs.microsoft.com/en-us/nuget/install-nuget-client-tools[NuGet]:
++
+[source,bash,subs="attributes"]
+----
+$ dotnet add package AeroGear.Mobile.Security --version {release-number}
```
Comment (pb82) [inline modules/ROOT/pages/security/coding.adoc]: where did you see `AeroGear.Mobile.Security`? i can't find that in https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/docs/modules/getting-started/pages/security.adoc

File: modules/ROOT/pages/security/coding.adoc (hunk 1 above)
Comment (aliok) [reply modules/ROOT/pages/security/coding.adoc]: I was also gonna suggest the same thing.

File: modules/ROOT/pages/security/coding.adoc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+This guide will help you to set up the {device-security-service} service SDK in your App.
+
+=== Importing the libraries
+
+[role="primary"]
+.Android
+
+****
+
+. Add `jcenter` to your project's *build.grade*:
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply modules/ROOT/pages/security/coding.adoc]: 🤦‍♂️

File: modules/ROOT/pages/security/coding.adoc (hunk 1 above)
Comment (josemigallas) [reply modules/ROOT/pages/security/coding.adoc]: Yes, I am editing it to look just like IDM

File: modules/ROOT/pages/security/coding.adoc (hunk 2 above)
Comment (josemigallas) [reply modules/ROOT/pages/security/coding.adoc]: I liked for it in Nuget: https://www.nuget.org/packages?q=Aerogear.mobile.security

**Comment by josemigallas** [comment]
@pb82 @aliok I updated `security/coding.adoc` and also a couple of minor touches to `keycloak/coding.adoc`.

File: modules/ROOT/pages/security/coding.adoc (hunk 2 above)
Comment (pb82) [reply modules/ROOT/pages/security/coding.adoc]: @ziccardi ping. Do we manually install `Aerogear.mobile.security`?

File: modules/ROOT/pages/security/ref_terminology.adoc [hunk 4]
```diff
@@ -0,0 +1,34 @@
+include::{partialsdir}/attributes.adoc[]
+
+//':context:' is a vital parameter. See: http://asciidoctor.org/docs/user-manual/#include-multiple
+:context: ref_terminology_{device-security-service}
+
+[id='{context}_ref_terminology']
+
+= Provided Self Defence Checks
+
+The following are the self defence checks provided by the SDK.
```
Comment (pb82) [inline modules/ROOT/pages/security/ref_terminology.adoc]: As discussed i think we should use a tabbed box here and copy over the content from the various platform SDKs.

File: modules/ROOT/nav.adoc [hunk 5]
```diff
@@ -20,6 +20,7 @@ include::{partialsdir}/attributes.adoc[]
 
 // SECURITY
 * xref:security/index.adoc[{device-security-service}]
+** xref:security/coding.adoc[Coding]
```
Comment (pb82) [inline modules/ROOT/nav.adoc]: There should not be a coding section

**Comment by josemigallas** [comment]
@finp @pb82 @jstaffor @jhellar 
*Device Trust Checks* doc is done, now it's time for the final review.

It's still WIP only because those 3 small items that are missing (due to lack of info). This will be addressed first thing tomorrow provided I am able to find the necessary info.

File: modules/ROOT/nav.adoc (hunk 5 above)
Comment (josemigallas) [reply modules/ROOT/nav.adoc]: After speaking with @finp we've decided to add it.

**Comment by StephenCoady** [comment]
@josemigallas to answer your question no, iOS doesn't have a developer mode check. Currently its debugger, jailbreak, device lock and emulator checks.

**Comment by josemigallas** [comment]
Resolved last TODO: task is currently in development -> https://issues.jboss.org/browse/AEROGEAR-2872.

@finp @pb82 I think this is ready to go.

**Pull request merged**